core/src/impl/Kokkos_Profiling_Interface.hpp

```diff
@@ -54,6 +54,17 @@
 #include <iostream>
 #include <cstdlib>
 
+// NOTE: in this Kokkos::Profiling block, do not define anything that shouldn't
+// exist should Profiling be disabled
+
+namespace Kokkos {
+namespace Profiling {
+enum DeviceType { Serial, OpenMP, Cuda, HIP, OpenMPTarget, HPX, ROCm, Threads };
```

Comment:
Any reason not to make it strongly typed?
```suggestion
enum struct DeviceType { Serial, OpenMP, Cuda, HIP, OpenMPTarget, HPX, ROCm, Threads };
```